core-rust/state-manager/src/mempool/priority_mempool.rs

```diff
@@ -299,8 +313,15 @@ impl PriorityMempool {
         let payload_hash = &data.transaction.notarized_transaction_hash();
         let intent_hash = &data.transaction.intent_hash();
 
+        let transaction_size = data.transaction.raw.0.len();
         self.remaining_transaction_count += 1;
-        self.remaining_total_transactions_size += data.transaction.raw.0.len() as u64;
+        self.remaining_total_transactions_size += transaction_size as u64;
+
+        // Update metrics
```

Comment:
I looked at the entire priority_mempool.rs and I am pretty convinced that we track all (2) add/remove operations.

But ofc, as always, I worry that a future refactor/feature will introduce another add/remove call, and we would miss a metric update.

Actually the same problem is with `proposal_priority_index` (well, probably harder to miss, since it would cause weird behavior right away, unlike metrics).

What do you think about introducing a `ListenableMap<K, V>`, which would allow for installing a `MapListener<K, V>`, which would get notified like `.onEntryAdded(K, V) / .onEntryRemoved(K, V)`?
(and using it for maintaining the `proposal_priority_index` and for the metric-bumping)

Reply:
I think it makes more sense to keep it as is and in future PRs touching this remind people to use only `remove_data` and `add_transaction`. There is also `intent_lookup` that needs maintaining.
If another way of adding transactions (without checking for priorities, sizes, etc.) will make sense we can split `add_transaction` at that point and keep only 2 "primitive/internal" methods that deal with all aux data/metrics.